origin-discovery/listener/listener.js

```diff
@@ -399,15 +424,44 @@ async function handleLog(log, rule, contractVersion, context) {
   }
 
   if (context.config.db) {
-    console.log(`Indexing listing in DB: id=${listingId}`)
-    await withRetrys(async () => {
-      await db.Listing.insert(
-        listingId,
-        userAddress,
-        ipfsHash,
-        listing.ipfs.data
-      )
-    })
+    if (LISTING_EVENTS.includes(rule.eventName)) {
+      console.log(`Indexing listing in DB: id=${listingId}`)
+      listingData = {
+        id: listingId,
+        status: listing.status,
+        sellerAddress: listing.seller.toLowerCase(),
+        data: listing,
+      }
+      if (rule.eventName === 'ListingCreated') {
+        listingData.createdAt = log.date
+      } else {
+        listingData.updatedAt = log.date
+      }
+      await withRetrys(async () => {
+        await db.Listing.insertOrUpdate(listingData)
```

Comment:
If I'm understanding correctly this represents a change from storing all the events for a listing to storing a single row per listing that gets updated with each new event. Are there any situations where we need the event history for a listing? Maybe if we had the complete event history we could recreate a listing at the time an offer was made? Might be something we need to think about later. I normally like an event sourcing style pattern for this sort of thing, but this may be cleaner and have better performance.

Not asking you to change anything, just thinking out loud. :grin:

Reply:
In order to create a Listing object, we need more than just events. We need the data from the smart contract as well as the IPFS data. From a latency standpoint, it would be pretty slow to fetch all that data for let's say 100 listings we want to show on Browse/Search page. That's why I went for storing the latest state of the Listing/Offer.

But I do agree that having a quick way to lookup events could be very useful in future. For example the DApp currently queries web3 for fetching data to use in the Notificationt tab.
Today I actually spend some time populating a Postgres "Event" table via the event-listener that will give us the event history. I'll send a PR once this PR is merged in since I need the sequelize scaffolding first :)